### PR TITLE
Azure IPI cluster in shared network

### PR DESCRIPTION
**Pull request by Amoghrd**
Changes to QE workflow for sharednetwork on the basis of https://github.com/openshift/release/pull/26947

### CODE REVIEW

**Comment by Amoghrd** [comment]
So, for this profile, Dev is using a vnet which is preserved. We as QE should create a RG, a VNET in it and then the remaining steps right?

File: ci-operator/step-registry/azure/provision/sharednetwork/azure-provision-sharednetwork-commands.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
```
Comment (jianlinliu) [inline ci-operator/step-registry/azure/provision/sharednetwork/azure-provision-sharednetwork-commands.sh:1]: I think we do not need this new ref, but we can improve the logic based on `ipi/conf/azure/sharednetwork/ipi-conf-azure-sharednetwork-commands.sh`.
Change the logic
```
cat > "${PATCH}" << EOF
platform:
  azure:
    networkResourceGroupName: os4-common
    virtualNetwork: do-not-delete-shared-vnet-${azure_region}
    controlPlaneSubnet: subnet-1
    computeSubnet: subnet-2
EOF
```
to 
```
if [ -f "${SHARED_DIR}/customer_vnet_subnets.yaml" ]; then
  <blablabla>
else
  cat > "${PATCH}" << EOF
platform:
  azure:
    networkResourceGroupName: os4-common
    virtualNetwork: do-not-delete-shared-vnet-${azure_region}
    controlPlaneSubnet: subnet-1
    computeSubnet: subnet-2
EOF
fi
```

**Pull request closed without merging**